### ISSUE TITLE
Suggestion: Make point change equal on both sides for victory/defeat

### ISSUE THREAD

**Issue by db0** [jrmitch120/ChallengeBoard#9]
With the current system, I believe it discourages players on the top to play players towards the bottom of the ranking, because the score change on a random loss (which can easily come from a bad draw) from a much lower-ranked player equal to about 3-4 games of victories against the same player (i.e. 25 points)

This means that the challengeboard punishes players who perform well, by making them lose a ton of points on a loss and thus either discouraging playing openly, or discouraging playing with lower-ranked players.

I believe scoring changes should be the same on both sides. This means a lower-ranked player losing to a higher ranked one should get the same +7. This will make people on the top more likely to play openly as a random loss won't drop one significantly.

**Comment by db0** [comment]
Another option would be to implement Glicko2?

**Comment by jrmitch120** [comment]
I looked at Glicko2 and was going to put it in.  In the end, I wasn't able to understand the math behind it well enough to get it to work and gave up.  If somebody wants to lend a hand, I'd be happy to get it in.

**Comment by apolon81** [comment]
I see couple of possibilities, which can be implemented selectively or simultaneously.

One thing could be adding possibility to define K-factors during the league setup. For example in case of 2-month long netrunner league, in the history of 3 leagues only once we have seen 1800 ELO achieved. I K buckets were, say "below 1600", "1600-1720", "above 1720" it would at least have a chance to kick in.

Second thing would be implementing Glicko and Glicko2, both of which can be calculated alongside with ELO in separate columns of the standings. During the league setup the owner would choose the official rating system to base the standings on, others could be displayed for curiosity sake. I've read Glicko pdf and can help with the implementation.

And last we have db0's proposal - PSR I guess?. Any official papers on that? I've found only links to LoL forum posts, not sure if this is mature enough to put effort in it..

**Comment by db0** [comment]
Well my main proposal was pretty simple. The gains/losses from large differences should always be small and the same for both sides.

**Comment by apolon81** [comment]
I'm not sure I like the same and small amount for either win or loss. It creates a "nothing to gain, nothing to lose" situation which is a bit anti-competitive. Also this might encourage top players to avoid each other and start "farming" on newcomers.

**Comment by db0** [comment]
The current situation is even worse imho. It gives an incentive to actively avoid playing lower rank players because a simple luck based loss is significantly more hurtful.

I don't think it will make players avoid playing if it was the same. There's just not enough players online at any point to be that picky. This way people can at least still game as much as they want without worrying that if they test their jank decks against a newbie it's going to cost them dearly.

**Comment by apolon81** [comment]
Agreed, but that's where adjustable K factors for ELO come to the rescue I would think?

**Comment by db0** [comment]
Tweaking the K-factor is not going to change that though. It's just that the loss to a lower ranked player is less, but then the spread of ratings is also lower.

**Comment by jrmitch120** [comment]
Tweaking the K factor can help here.  Right now it scales thusly:

- 32 KFactor - Starting Rating to Starting Rating * 1.3125
- 24 KFactor - Starting Rating * 1.3125 to Starting Rating * 1.5
- 16 KFactor - Over Starting Rating * 1.5

This basically means that as your rank rises, your KFactor goes down, decreasing your potential rating swing.

So using a 1500 base, we're looking at breaks of 1968, 2250, and 2250+.  This clearly is shit based on my cursory look at boards.  I don't know if I've ever seen a 1900+, so we're basically dealing with a constant 32K, which isn't doing anything.

I would also certainly be interested in plopping in Glicko2 or another different ratings system.  As Apolon suggests, this could be calculated alongside, or I would prefer, chosen up front by the board creator.  

On another note, I think another real challenge is to figure out a way to remove the stigma that higher vs lower ranked players has.  Regardless of whatever system is in place.  I think that is going to be an issue, even if it isn't based in fact.

I know in video gaming, often other player ratings are hidden from public view.  I wonder if that could help here.  Maybe a configurable option to hide overall ratings until the end (you can still see your own throughout).

**Comment by sandroos** [comment]
A pair of games in Netrunner (corp+runner) could also be treated as a single 'game' in (ELO terms)[http://www.chess-mind.com/en/elo-system], i.e., in the ELO formula `S=0` if the game is a draw. While the higher ranked player would still lose some points, at least the changes in players' ratings would not depend on the order in which the games were reported.

If I wanted to, say, implement Glicko, how do I run the demo site / test the changes? I have Linux on my laptop.

**Comment by sandroos** [comment]
>So using a 1500 base, we're looking at breaks of 1968, 2250, and 2250+. This clearly is shit based on my cursory look at boards. I don't know if I've ever seen a 1900+, so we're basically dealing with a constant 32K, which isn't doing anything.

In Stimhack league players rarely reach 1800, so 1600/1700/1800 might be more appropriate breakpoints. Ideally these values should be set by the board creator.

**Comment by jrmitch120** [comment]
@sandroos In theory, you can get it running using mono develop, although I myself haven't had experience getting an MVC site up and running through it.  Another option might be to virutal box a Windows machine if you have a copy of the OS.

I was thinking about board adjustable k's.  Seems like a small step in the right direction.

**Comment by apolon81** [comment]
I guess next steps are adjustable K's and Glicko2. I'll grab Glicko2.

@jrmitch120 if you prefer to set the algorithm upfront and calculate only this one, it's ok with me also.

**Comment by jrmitch120** [comment]
@apolon81 Glicko2 will garner you some serious manlove ;)  I'll see if I can pluck off a couple other things that bug me right now.  The ability to withdraw a submission probably for starts.
